actors/market/tests/publish_storage_deals_failures.rs

```diff
@@ -203,4 +208,268 @@ mod publish_storage_deals_failures {
+        let sig = Signature::new_bls("does not matter".as_bytes().to_vec());
+        let params = PublishStorageDealsParams {
+            deals: vec![ClientDealProposal { proposal: deal, client_signature: sig }],
+        };
+        let w = Address::new_id(1000);
+        rt.set_caller(*MINER_ACTOR_CODE_ID, w);
+        rt.expect_validate_caller_type(vec![*ACCOUNT_ACTOR_CODE_ID, *MULTISIG_ACTOR_CODE_ID]);
+        expect_abort(
+            ExitCode::USR_FORBIDDEN,
+            rt.call::<MarketActor>(
+                Method::PublishStorageDeals as u64,
+                &RawBytes::serialize(params).unwrap(),
+            ),
+        );
+        check_state(&rt);
+    }
+
+    #[test]
+    fn fail_when_no_deals_in_params() {
+        let mut rt = setup();
+        let params = PublishStorageDealsParams { deals: vec![] };
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, WORKER_ADDR);
+        rt.expect_validate_caller_type(vec![*ACCOUNT_ACTOR_CODE_ID, *MULTISIG_ACTOR_CODE_ID]);
+        expect_abort(
+            ExitCode::USR_ILLEGAL_ARGUMENT,
+            rt.call::<MarketActor>(
+                Method::PublishStorageDeals as u64,
+                &RawBytes::serialize(params).unwrap(),
+            ),
+        );
+        check_state(&rt);
+    }
+
+    #[test]
+    fn fail_to_resolve_provider_address() {
+        let start_epoch = 10;
+        let end_epoch = start_epoch + 200 * EPOCHS_IN_DAY;
+
+        let mut rt = setup();
+        let mut deal = generate_deal_proposal(CLIENT_ADDR, PROVIDER_ADDR, start_epoch, end_epoch);
+        deal.provider = new_bls_addr(100);
+
+        let sig = Signature::new_bls("does not matter".as_bytes().to_vec());
+        let params = PublishStorageDealsParams {
+            deals: vec![ClientDealProposal { proposal: deal, client_signature: sig }],
+        };
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, WORKER_ADDR);
+        rt.expect_validate_caller_type(vec![*ACCOUNT_ACTOR_CODE_ID, *MULTISIG_ACTOR_CODE_ID]);
+        expect_abort(
+            ExitCode::USR_NOT_FOUND,
+            rt.call::<MarketActor>(
+                Method::PublishStorageDeals as u64,
+                &RawBytes::serialize(params).unwrap(),
+            ),
+        );
+        check_state(&rt);
+    }
+
+    #[test]
+    fn caller_is_not_the_same_as_the_worker_address_for_miner() {
+        let start_epoch = 10;
+        let end_epoch = start_epoch + 200 * EPOCHS_IN_DAY;
+
+        let mut rt = setup();
+        let deal = generate_deal_proposal(CLIENT_ADDR, PROVIDER_ADDR, start_epoch, end_epoch);
+        let sig = Signature::new_bls("does not matter".as_bytes().to_vec());
+        let params = PublishStorageDealsParams {
+            deals: vec![ClientDealProposal { proposal: deal, client_signature: sig }],
+        };
+
+        rt.expect_validate_caller_type(vec![*ACCOUNT_ACTOR_CODE_ID, *MULTISIG_ACTOR_CODE_ID]);
+        expect_provider_control_address(&mut rt, PROVIDER_ADDR, OWNER_ADDR, Address::new_id(999));
```

Comment:
The go test is confusing here.  The actual worker address is 999 but this test calls the caller address worker addr.  The easiest fix is probably to set the caller to 999 and keep the worker address in the control address mock return.  Alternatively you can create a local caller address and set it to something other than the worker.